### PR TITLE
chore: remove show receive amount in select provider feature gate

### PR DESCRIPTION
**Pull request by MuckT**
### Description

The `show_receive_amount_in_select_provider` feature gate has been rolled out to 100% since v1.62.0. This is to remove the code path for when that gate returned false.

### Test plan

- Tested locally on iOS
- Unit Tests updated
- E2E tests updated

### Related issues

- Fixes ACT-1076

### Backwards compatibility

Yes

### CODE REVIEW

File: e2e/src/usecases/OnRamps.js [hunk 1]
```diff
@@ -42,11 +42,6 @@ export default onRamps = () => {
           device.getPlatform() === 'ios'
             ? cardProviders.label.split(' ')[0]
             : cardProviders.text.split(' ')[0]
-        await element(by.id('Card/section')).tap()
```
Comment (MuckT) [inline e2e/src/usecases/OnRamps.js:45]: This is removed as the section now starts as expanded.

File: e2e/src/usecases/OnRamps.js [hunk 2]
```diff
@@ -42,11 +42,6 @@ export default onRamps = () => {
...
-        // Check that best rate is displayed first
-        await expect(
-          element(by.id('Card/provider-0').withDescendant(by.id('Card/bestRate')))
```
Comment (MuckT) [inline e2e/src/usecases/OnRamps.js:48]: Best Rate no longer displayed.

File: src/fiatExchanges/FiatExchangeAmount.test.tsx [hunk 3]
```diff
@@ -243,51 +243,40 @@ describe('FiatExchangeAmount cashOut', () => {
...
         <FiatExchangeAmount {...mockScreenPropsCelo} />
       </Provider>
     )
     expect(getByText('amount (CELO)')).toBeTruthy()
-    expect(getElementText(getByTestId('LineItemRowTitle/subtotal'))).toBe('CELO @ $5.00')
-    expect(getElementText(getByTestId('LineItemRow/subtotal'))).toBe('$0.00')
   })
 
-  it('displays correctly when the SHOW_RECEIVE_AMOUNT_IN_SELECT_PROVIDER feature flag is on', () => {
+  it('displays correctly', () => {
```
Comment (jophish) [inline src/fiatExchanges/FiatExchangeAmount.test.tsx:272]: This test is now identical to the `'displays correctly for cUSD when local currency is USD'` one, right? Can prolly remove this one now.

**Review by jophish — COMMENTED** [review]
Looks like a couple of e2e tests for iOS are failing in ` e2e/src/AccountManagement.spec.js`

File: src/fiatExchanges/FiatExchangeAmount.test.tsx (hunk 3 above)
Comment (MuckT) [reply src/fiatExchanges/FiatExchangeAmount.test.tsx]: Fixed in 0efe49eea5a1dd1a63affd881569a971f3912391!

**Comment by MuckT** [comment]
@jophish looks like the failing e2e test was due to a transient failure with our currency API.

**Review by jh2oman — APPROVED** [review]

**Pull request merged**